### PR TITLE
Make role picker nicer

### PR DESCRIPTION
**Pull request by jakob223**
I'm concerned whether the animations are too slow / too choppy. Possibly due to me nesting 2 react-springs inside one another. Interested to hear thoughts on improving that in particular.

cc @keybase/y2ksquad

### CODE REVIEW

File: shared/teams/add-members-wizard/confirm.tsx [hunk 1]
```diff
@@ -185,53 +185,39 @@ const AddMoreMembers = () => {
...
-const RoleSelector = () => {
+type RoleType = 'owner' | 'admin' | 'reader' | 'writer' | 'setIndividually'
+const RoleSelector = ({memberCount}: {memberCount: number}) => {
   const dispatch = Container.useDispatch()
   const [showingMenu, setShowingMenu] = React.useState(false)
-  const storeRole = Container.useSelector(s => s.teams.addMembersWizard.role)
-  const [role, setRole] = React.useState(storeRole)
-  const onSelectRole = newRole => setRole(newRole)
-  const onConfirmRole = newRole => {
+  // as because bot roles
```
Comment (buoyad) [inline shared/teams/add-members-wizard/confirm.tsx:192]: then i think the type belongs in `Types` and on the `addMembersWizard` state

File: shared/teams/add-members-wizard/confirm.tsx [hunk 2]
```diff
@@ -185,53 +185,39 @@ const AddMoreMembers = () => {
...
-                Or{' '}
-                <Kb.Text type="BodySmallPrimaryLink" onClick={onSetIndividually}>
-                  set roles individually
-                </Kb.Text>
-              </Kb.Text>
-            </Kb.Box2>
-          )
-        }
+        confirmLabel="Save"
+        includeSetIndividually={!Styles.isMobile}
```
Comment (buoyad) [inline shared/teams/add-members-wizard/confirm.tsx:212]: ✨ check # of selected members > 1

File: shared/teams/add-members-wizard/confirm.tsx [hunk 3]
```diff
@@ -324,6 +312,8 @@ const AddingMember = (props: Types.AddingMember & {lastMember?: boolean}) => {
         {showDropdown && (
           <FloatingRolePicker
             open={showingMenu}
+            presetRole={individualRole}
+            onCancel={individualRole === rolePickerRole ? () => setShowingMenu(false) : undefined}
```
Comment (buoyad) [inline shared/teams/add-members-wizard/confirm.tsx:316]: :sheep: would it be much harder to close the menu regardless? feels weird to me for it to close sometimes

File: shared/teams/role-picker/index.tsx [hunk 4]
```diff
@@ -56,21 +69,47 @@ const RoleRow = (p: RoleRowProps) => (
...
+  return (
+    <Kb.Box2
+      direction="vertical"
+      fullWidth={true}
+      alignItems="flex-start"
+      style={p.selected ? styles.rowSelected : styles.row}
+    >
+      {p.disabledReason ? (
+        <Kb.WithTooltip tooltip={p.disabledReason} showOnPressMobile={true}>
+          {/* Why is this tooltip the wrong color */}
```
Comment (buoyad) [inline shared/teams/role-picker/index.tsx:96]: Can we fix?

File: shared/teams/role-picker/index.tsx [hunk 5]
```diff
@@ -156,36 +201,36 @@ const roleAbilities = (
...
-      <Kb.Text type="BodySmall">{ability}</Kb.Text>
+      <Kb.Text type="BodySmall" style={canDo ? styles.canText : undefined}>
+        {ability}
+      </Kb.Text>
     </Kb.Box2>
   ))
 }
 
-const roleElementHelper = (selectedRole: Role | null) =>
+const roleElementHelper = <IncludeSetIndividually extends boolean>(
```
Comment (buoyad) [inline shared/teams/role-picker/index.tsx:217]: I don't think this being generic is doing anything?

File: shared/teams/role-picker/index.tsx [hunk 6]
```diff
@@ -225,45 +258,69 @@ const footerButtonsHelper = (
   </Kb.ButtonBar>
 )
 
-const confirmLabelHelper = (presetRole: Role | null, selectedRole: Role | null): string => {
+const confirmLabelHelper = <IncludeSetIndividually extends boolean>(
```
Comment (buoyad) [inline shared/teams/role-picker/index.tsx:261]: Same here

File: shared/teams/add-members-wizard/confirm.tsx (hunk 3 above)
Comment (jakob223) [reply shared/teams/add-members-wizard/confirm.tsx:316]: Hmm. I think a lot of places misuse the props of the role picker and change their internal state as soon as a new role is selected, so cancelling that seems ill-defined.

File: shared/wallets/wallet/settings/index.stories.tsx [hunk 7]
```diff
@@ -48,7 +48,10 @@ const externalPartner = {
   title: 'Example title.',
   url: 'https://example.com/%{accountId}',
 }
-const externalPartners = [externalPartner, {...externalPartner, showDivider: true}]
+const externalPartners = [
+  externalPartner,
+  {...externalPartner, showDivider: true, url: 'https://differenturl.com/%{accountId'},
```
Comment (buoyad) [inline shared/wallets/wallet/settings/index.stories.tsx:53]: ?

Comment (jakob223) [reply shared/wallets/wallet/settings/index.stories.tsx:53]: duplicate key error

**Pull request merged**